builtin/mainmenu/dlg_settings_advanced.lua

```diff
@@ -667,13 +667,17 @@ local function create_change_setting_formspec(dialogdata)
 		height = height + 1.1
 
 	elseif setting.type == "flags" then
-		local enabled_flags = flags_to_table(get_current_value(setting))
+		local current_flags = flags_to_table(get_current_value(setting))
 		local flags = {}
-		for _, name in ipairs(enabled_flags) do
+		for _, name in ipairs(current_flags) do
 			-- Index by name, to avoid iterating over all enabled_flags for every possible flag.
-			flags[name] = true
+			if name:sub(1, 2) == "no" then
+				flags[name:sub(3)] = false
+			else
+				flags[name] = true
+			end
 		end
-		local flags_count = #setting.possible
+		local flags_count = #setting.possible / 2
 		local max_height = flags_count / 4
```

Comment:
```suggestion
		local max_height = math.ceil(flags_count / 2) / 2
```
To fix the height when there's an odd number of flags.

Reply:
Fixed.